src/core/ipe-setup.c

```diff
@@ -0,0 +1,122 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "conf-files.h"
+#include "copy.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "ipe-setup.h"
+#include "path-util.h"
+
+#define IPE_SECFS_DIR "/sys/kernel/security/ipe"
+#define IPE_SECFS_NEW_POLICY IPE_SECFS_DIR "/new_policy"
+#define IPE_SECFS_POLICIES IPE_SECFS_DIR "/policies/"
+
+int ipe_setup(void) {
+#if ENABLE_IPE
+        _cleanup_strv_free_ char **policies = NULL;
+        int r;
+
+        /* Very quick smoke tests first: this is in the citical, sequential boot path, and in most cases it
+         * is unlikely this will be configured, so do the fastest existence checks first and immediately
+         * return if there's nothing to do. */
+
+        if (access(IPE_SECFS_DIR, F_OK) < 0) {
+                log_debug_errno(errno, "IPE support is disabled in the kernel, ignoring: %m");
+                return 0;
+        }
+
+        bool found = false;
+        STRV_FOREACH(p, CONF_PATHS_NULSTR("ipe"))
+                if (dir_is_empty(p, /* ignore_hidden_or_backup= */ true) == 0) {
+                        found = true;
+                        break;
+                }
+        if (!found) {
+                log_debug("No IPE policy files found, ignoring.");
+                return 0;
+        }
+
+        r = conf_files_list_nulstr(
+                        &policies,
+                        ".p7b",
+                        /* root= */ NULL,
+                        CONF_FILES_REGULAR|CONF_FILES_FILTER_MASKED,
+                        CONF_PATHS_NULSTR("ipe"));
+        if (r < 0)
+                return log_error_errno(r, "Failed to assemble list of IPE policies: %m");
+
+        STRV_FOREACH(policy, policies) {
+                _cleanup_free_ char *policy_name = NULL, *file_name = NULL, *output_path = NULL, *activate_path = NULL;
+                _cleanup_close_ int input = -EBADF, output = -EBADF;
+                const char *suffix;
+
+                r = path_extract_filename(*policy, &file_name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to extract filename from IPE policy path %s: %m", *policy);
+
+                suffix = endswith(file_name, ".p7b");
+                if (!suffix)
+                        continue;
+
+                policy_name = strndup(file_name, suffix - file_name);
+                if (!policy_name)
+                        return log_oom();
+
+                if (!filename_is_valid(policy_name))
+                        return log_error_errno(
+                                        EINVAL,
+                                        "Invalid IPE policy name %s",
+                                        policy_name);
```

Comment:
I'd just warn about this, and skip it, i.e. `continue`

Reply:
I'd rather fail hard, to make it very clear and impossible to miss when accidental mistakes in installing policies are done, given if no policy is loaded, then the system is open